[Build] windows minimal noexception builds fail to compile

### Describe the issue

Windows minimal noexception builds fail to compile on recent `main` branch commit d2188fbff99a8f06982d902e89e6de139d4a627e.
Unknown when the build was broken or by whom.

Linux minimal noexception builds seem to work.

### Urgency

Can lead to PR churn. One of the CI tests configured for a PR is a minimal noexception build. Yet, if we (on windows) can not compile and test it locally, then we have to use the CI for testing of this specific min+noexcept build...leading to churn.

### Target platform

Windows

### Build script

```
.\build.bat --cmake_generator "Visual Studio 16 2019" --config Debug --build_shared_lib --parallel --minimal_build --disable_exceptions
```

### Error / output

```
...
...
C:\repos-nobackup\onnxruntime\build\Windows\Debug\_deps\googletest-src\googlemock\include\gmock/gmock-matchers.h(2438,1
): error C2678: binary '==': no operator found which takes a left-hand operand of type 'const gsl::span<const int8_t,18
446744073709551615>' (or there is no acceptable conversion) (compiling source file C:\repos-nobackup\onnxruntime\onnxru
ntime\test\util\test_utils.cc) [C:\repos-nobackup\onnxruntime\build\Windows\Debug\onnxruntime_test_utils.vcxproj]
C:\repos-nobackup\onnxruntime\build\Windows\Debug\_deps\googletest-src\googletest\include\gtest/gtest.h(1362,13): messa
ge : could be 'bool testing::internal::operator ==(testing::internal::faketype,testing::internal::faketype)' (compiling
 source file C:\repos-nobackup\onnxruntime\onnxruntime\test\util\test_utils.cc) [C:\repos-nobackup\onnxruntime\build\Wi
ndows\Debug\onnxruntime_test_utils.vcxproj]
C:\repos-nobackup\onnxruntime\build\Windows\Debug\_deps\googletest-src\googlemock\include\gmock/gmock-matchers.h(2438,1
): message : while trying to match the argument list '(const gsl::span<const int8_t,18446744073709551615>, const gsl::s
pan<const int8_t,18446744073709551615>)' (compiling source file C:\repos-nobackup\onnxruntime\onnxruntime\test\util\tes
t_utils.cc) [C:\repos-nobackup\onnxruntime\build\Windows\Debug\onnxruntime_test_utils.vcxproj]
C:\repos-nobackup\onnxruntime\build\Windows\Debug\_deps\googletest-src\googletest\include\gtest/gtest-matchers.h(683):
message : see reference to function template instantiation 'bool testing::internal::ContainerEqMatcher<gsl::span<const
int8_t,18446744073709551615>>::MatchAndExplain<T>(const LhsContainer &,testing::MatchResultListener *) const' being com
piled [C:\repos-nobackup\onnxruntime\build\Windows\Debug\onnxruntime_test_utils.vcxproj]
          with
          [
              T=gsl::span<const int8_t,18446744073709551615>,
              LhsContainer=gsl::span<const int8_t,18446744073709551615>
          ] (compiling source file C:\repos-nobackup\onnxruntime\onnxruntime\test\util\test_utils.cc)
C:\repos-nobackup\onnxruntime\build\Windows\Debug\_deps\googletest-src\googletest\include\gtest/gtest-matchers.h(683):
message : see reference to function template instantiation 'bool testing::internal::ContainerEqMatcher<gsl::span<const
int8_t,18446744073709551615>>::MatchAndExplain<T>(const LhsContainer &,testing::MatchResultListener *) const' being com
piled [C:\repos-nobackup\onnxruntime\build\Windows\Debug\onnxruntime_test_utils.vcxproj]
          with
          [
              T=gsl::span<const int8_t,18446744073709551615>,
              LhsContainer=gsl::span<const int8_t,18446744073709551615>
          ] (compiling source file C:\repos-nobackup\onnxruntime\onnxruntime\test\util\test_utils.cc)
C:\repos-nobackup\onnxruntime\build\Windows\Debug\_deps\googletest-src\googletest\include\gtest/gtest-matchers.h(682):
message : while compiling class template member function 'bool testing::PolymorphicMatcher<testing::internal::Container
EqMatcher<gsl::span<const int8_t,18446744073709551615>>>::MonomorphicImpl<T>::MatchAndExplain(T,testing::MatchResultLis
tener *) const' [C:\repos-nobackup\onnxruntime\build\Windows\Debug\onnxruntime_test_utils.vcxproj]
          with
          [
              T=const gsl::span<const int8_t,18446744073709551615> &
          ] (compiling source file C:\repos-nobackup\onnxruntime\onnxruntime\test\util\test_utils.cc)
C:\repos-nobackup\onnxruntime\build\Windows\Debug\_deps\googletest-src\googletest\include\gtest/gtest-matchers.h(667):
message : see reference to class template instantiation 'testing::PolymorphicMatcher<testing::internal::ContainerEqMatc
her<gsl::span<const int8_t,18446744073709551615>>>::MonomorphicImpl<T>' being compiled [C:\repos-nobackup\onnxruntime\b
uild\Windows\Debug\onnxruntime_test_utils.vcxproj]
          with
          [
              T=const gsl::span<const int8_t,18446744073709551615> &
          ] (compiling source file C:\repos-nobackup\onnxruntime\onnxruntime\test\util\test_utils.cc)
C:\repos-nobackup\onnxruntime\build\Windows\Debug\_deps\googletest-src\googlemock\include\gmock/gmock-matchers.h(370):
message : see reference to function template instantiation 'testing::PolymorphicMatcher<testing::internal::ContainerEqM
atcher<gsl::span<const int8_t,18446744073709551615>>>::operator testing::Matcher<const T &>(void) const<const T&>' bein
g compiled [C:\repos-nobackup\onnxruntime\build\Windows\Debug\onnxruntime_test_utils.vcxproj]
          with
          [
              T=gsl::span<const int8_t,18446744073709551615>
          ] (compiling source file C:\repos-nobackup\onnxruntime\onnxruntime\test\util\test_utils.cc)
C:\repos-nobackup\onnxruntime\build\Windows\Debug\_deps\googletest-src\googlemock\include\gmock/gmock-matchers.h(370):
message : see reference to function template instantiation 'testing::PolymorphicMatcher<testing::internal::ContainerEqM
atcher<gsl::span<const int8_t,18446744073709551615>>>::operator testing::Matcher<const T &>(void) const<const T&>' bein
g compiled [C:\repos-nobackup\onnxruntime\build\Windows\Debug\onnxruntime_test_utils.vcxproj]
          with
          [
              T=gsl::span<const int8_t,18446744073709551615>
          ] (compiling source file C:\repos-nobackup\onnxruntime\onnxruntime\test\util\test_utils.cc)
C:\repos-nobackup\onnxruntime\build\Windows\Debug\_deps\googletest-src\googlemock\include\gmock/gmock-matchers.h(354):
message : see reference to function template instantiation 'testing::Matcher<const T &> testing::internal::MatcherCastI
mpl<const gsl::span<const int8_t,18446744073709551615>&,M>::CastImpl<false>(const M &,std::true_type,std::integral_cons
tant<bool,false>)' being compiled [C:\repos-nobackup\onnxruntime\build\Windows\Debug\onnxruntime_test_utils.vcxproj]
          with
          [
              T=gsl::span<const int8_t,18446744073709551615>,
              M=testing::PolymorphicMatcher<testing::internal::ContainerEqMatcher<gsl::span<const int8_t,18446744073709
  551615>>>
          ] (compiling source file C:\repos-nobackup\onnxruntime\onnxruntime\test\util\test_utils.cc)
C:\repos-nobackup\onnxruntime\build\Windows\Debug\_deps\googletest-src\googlemock\include\gmock/gmock-matchers.h(352):
message : see reference to function template instantiation 'testing::Matcher<const T &> testing::internal::MatcherCastI
mpl<const gsl::span<const int8_t,18446744073709551615>&,M>::CastImpl<false>(const M &,std::true_type,std::integral_cons
tant<bool,false>)' being compiled [C:\repos-nobackup\onnxruntime\build\Windows\Debug\onnxruntime_test_utils.vcxproj]
          with
          [
              T=gsl::span<const int8_t,18446744073709551615>,
              M=testing::PolymorphicMatcher<testing::internal::ContainerEqMatcher<gsl::span<const int8_t,18446744073709
  551615>>>
          ] (compiling source file C:\repos-nobackup\onnxruntime\onnxruntime\test\util\test_utils.cc)
C:\repos-nobackup\onnxruntime\build\Windows\Debug\_deps\googletest-src\googlemock\include\gmock/gmock-matchers.h(338):
message : while compiling class template member function 'testing::Matcher<const T &> testing::internal::MatcherCastImp
l<const gsl::span<const int8_t,18446744073709551615>&,M>::Cast(const M &)' [C:\repos-nobackup\onnxruntime\build\Windows
\Debug\onnxruntime_test_utils.vcxproj]
          with
          [
              T=gsl::span<const int8_t,18446744073709551615>,
              M=testing::PolymorphicMatcher<testing::internal::ContainerEqMatcher<gsl::span<const int8_t,18446744073709
  551615>>>
          ] (compiling source file C:\repos-nobackup\onnxruntime\onnxruntime\test\util\test_utils.cc)
C:\repos-nobackup\onnxruntime\build\Windows\Debug\_deps\googletest-src\googlemock\include\gmock/gmock-matchers.h(512):
message : see reference to function template instantiation 'testing::Matcher<const T &> testing::internal::MatcherCastI
mpl<const gsl::span<const int8_t,18446744073709551615>&,M>::Cast(const M &)' being compiled [C:\repos-nobackup\onnxrunt
ime\build\Windows\Debug\onnxruntime_test_utils.vcxproj]
          with
          [
              T=gsl::span<const int8_t,18446744073709551615>,
              M=testing::PolymorphicMatcher<testing::internal::ContainerEqMatcher<gsl::span<const int8_t,18446744073709
  551615>>>
          ] (compiling source file C:\repos-nobackup\onnxruntime\onnxruntime\test\util\test_utils.cc)
C:\repos-nobackup\onnxruntime\build\Windows\Debug\_deps\googletest-src\googlemock\include\gmock/gmock-matchers.h(512):
message : see reference to class template instantiation 'testing::internal::MatcherCastImpl<T,M>' being compiled [C:\re
pos-nobackup\onnxruntime\build\Windows\Debug\onnxruntime_test_utils.vcxproj]
          with
          [
              T=const gsl::span<const int8_t,18446744073709551615> &,
              M=testing::PolymorphicMatcher<testing::internal::ContainerEqMatcher<gsl::span<const int8_t,18446744073709
  551615>>>
          ] (compiling source file C:\repos-nobackup\onnxruntime\onnxruntime\test\util\test_utils.cc)
C:\repos-nobackup\onnxruntime\build\Windows\Debug\_deps\googletest-src\googlemock\include\gmock/gmock-matchers.h(519):
message : see reference to function template instantiation 'testing::Matcher<const T &> testing::MatcherCast<const gsl:
:span<const int8_t,18446744073709551615>&,M>(const M &)' being compiled [C:\repos-nobackup\onnxruntime\build\Windows\De
bug\onnxruntime_test_utils.vcxproj]
          with
          [
              T=gsl::span<const int8_t,18446744073709551615>,
              M=testing::PolymorphicMatcher<testing::internal::ContainerEqMatcher<gsl::span<const int8_t,18446744073709
  551615>>>
          ] (compiling source file C:\repos-nobackup\onnxruntime\onnxruntime\test\util\test_utils.cc)
C:\repos-nobackup\onnxruntime\build\Windows\Debug\_deps\googletest-src\googlemock\include\gmock/gmock-matchers.h(1587):
 message : see reference to function template instantiation 'testing::Matcher<const T &> testing::SafeMatcherCast<const
 T&,M>(const M &)' being compiled [C:\repos-nobackup\onnxruntime\build\Windows\Debug\onnxruntime_test_utils.vcxproj]
          with
          [
              T=gsl::span<const int8_t,18446744073709551615>,
              M=testing::PolymorphicMatcher<testing::internal::ContainerEqMatcher<gsl::span<const int8_t,18446744073709
  551615>>>
          ] (compiling source file C:\repos-nobackup\onnxruntime\onnxruntime\test\util\test_utils.cc)
C:\repos-nobackup\onnxruntime\onnxruntime\test\util\test_utils.cc(46): message : see reference to function template ins
tantiation 'testing::AssertionResult testing::internal::PredicateFormatterFromMatcher<testing::PolymorphicMatcher<testi
ng::internal::ContainerEqMatcher<gsl::span<const int8_t,18446744073709551615>>>>::operator ()<gsl::span<const int8_t,18
446744073709551615>>(const char *,const T &) const' being compiled [C:\repos-nobackup\onnxruntime\build\Windows\Debug\o
nnxruntime_test_utils.vcxproj]
          with
          [
              T=gsl::span<const int8_t,18446744073709551615>
          ]
C:\repos-nobackup\onnxruntime\onnxruntime\test\util\test_utils.cc(46): message : see reference to function template ins
tantiation 'testing::AssertionResult testing::internal::PredicateFormatterFromMatcher<testing::PolymorphicMatcher<testi
ng::internal::ContainerEqMatcher<gsl::span<const int8_t,18446744073709551615>>>>::operator ()<gsl::span<const int8_t,18
446744073709551615>>(const char *,const T &) const' being compiled [C:\repos-nobackup\onnxruntime\build\Windows\Debug\o
nnxruntime_test_utils.vcxproj]
          with
          [
              T=gsl::span<const int8_t,18446744073709551615>
          ]
  op_kernel.cc
...
...
     Creating library C:/repos-nobackup/onnxruntime/build/Windows/Debug/Debug/onnxruntime.lib and object C:/repos-nobac
  kup/onnxruntime/build/Windows/Debug/Debug/onnxruntime.exp
  onnxruntime.vcxproj -> C:\repos-nobackup\onnxruntime\build\Windows\Debug\Debug\onnxruntime.dll
Traceback (most recent call last):
  File "C:\repos-nobackup\onnxruntime\tools\ci_build\build.py", line 2865, in <module>
    sys.exit(main())
  File "C:\repos-nobackup\onnxruntime\tools\ci_build\build.py", line 2763, in main
    build_targets(args, cmake_path, build_dir, configs, num_parallel_jobs, args.target)
  File "C:\repos-nobackup\onnxruntime\tools\ci_build\build.py", line 1371, in build_targets
    run_subprocess(cmd_args, env=env)
  File "C:\repos-nobackup\onnxruntime\tools\ci_build\build.py", line 752, in run_subprocess
    return run(*args, cwd=cwd, capture_stdout=capture_stdout, shell=shell, env=my_env)
  File "C:\repos-nobackup\onnxruntime\tools\python\util\run.py", line 49, in run
    completed_process = subprocess.run(
  File "C:\python39\lib\subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['C:\\Program Files\\cmake-3.25.0\\bin\\cmake.EXE', '--build', 'C:\\repos-nobackup\\onnxruntime\\\\build\\Windows\\Debug', '--config', 'Debug', '--', '/maxcpucount:16', '/nodeReuse:False']' returned non-zero exit status 1.
```

### Visual Studio Version

VS2019 v16.11.20

### GCC / Compiler Version

_No response_

--- a/onnxruntime/test/util/test_utils.cc
+++ b/onnxruntime/test/util/test_utils.cc
@@ -43,7 +43,7 @@ static void VerifyOutputs(const std::vector<std::string>& output_names,
             << " mismatch for " << output_names[i];
         break;
       case ONNX_NAMESPACE::TensorProto_DataType_INT8:
-        EXPECT_THAT(ltensor.DataAsSpan<int8_t>(), ::testing::ContainerEq(rtensor.DataAsSpan<int8_t>()))
+        EXPECT_TRUE(SpanEq(ltensor.DataAsSpan<int8_t>(), rtensor.DataAsSpan<int8_t>()))
             << " mismatch for " << output_names[i];
         break;
       case ONNX_NAMESPACE::TensorProto_DataType_FLOAT: {

